geoserver/src/eumetsat/pinning/src/main/java/org/geoserver/eumetsat/pinning/ViewsEvaluator.java

```diff
@@ -0,0 +1,543 @@
+                    continue;
+                }
+                if (newRight.isAfter(previousOrigRight)) {
+                    if (newRight.isAfter(previousMainRight)) {
+                        if (previousMainLeft.isAfter(previousOrigLeft) || timeUnchanged) {
+                            setPinLayerRange(
+                                    previousMainRight.plus(Duration.ofSeconds(1)),
+                                    newRight,
+                                    mappedLayer,
+                                    true);
+                        } else {
+                            setPinLayerRange(
+                                    previousMainLeft,
+                                    previousOrigLeft.minus(Duration.ofSeconds(1)),
+                                    mappedLayer,
+                                    false);
+                            setPinLayerRange(
+                                    previousOrigRight.plus(Duration.ofSeconds(1)),
+                                    newRight,
+                                    mappedLayer,
+                                    true);
+                        }
+                    } else {
+                        setPinLayerRange(
+                                newLeft.plus(Duration.ofSeconds(1)),
+                                previousMainRight,
+                                mappedLayer,
+                                false);
+                    }
+
+                } else {
+                    if (newLeft.isBefore(previousMainLeft)) {
+                        if (previousMainRight.isBefore(previousOrigRight) || timeUnchanged) {
+                            setPinLayerRange(
+                                    newLeft,
+                                    previousOrigLeft.minus(Duration.ofSeconds(1)),
+                                    mappedLayer,
+                                    true);
+                        } else {
+                            setPinLayerRange(
+                                    previousOrigRight.plus(Duration.ofSeconds(1)),
+                                    previousMainRight,
+                                    mappedLayer,
+                                    false);
+                            setPinLayerRange(
+                                    newLeft,
+                                    previousOrigLeft.minus(Duration.ofSeconds(1)),
+                                    mappedLayer,
+                                    true);
+                        }
+                    } else {
+                        setPinLayerRange(
+                                previousMainLeft,
+                                newLeft.minus(Duration.ofSeconds(1)),
+                                mappedLayer,
+                                false);
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean isExtendingWindow(Instant originalTime, Instant viewTime) {
```

Comment:
Note that this has been already updated to reflect this check:
|time_original - t1| <= Pinning Window

Reply:
I have added control of the windows extension against the driving layer.